### DISCUSSION TITLE
How to model a "similar movies" relationship?

### DISCUSSION THREAD

**Discussion by brielov** [prisma/prisma#9164] [Q&A] [answered]
Sorry if this is a stupid question, but I'm trying to model a movies database and each move can have many recommended movies, but when I try to do a self-to-many relation it gives me an error saying:

```
Error validating field `movie` in model `Movie`: The relation field `movie` on Model `Movie` is missing an opposite relation field on the model `Movie`. Either run `prisma format` or add it manually.
```

An example model:

```prisma
model Movie {
  id String @id @default(cuid())
  title String
  similar Movie[] @relation(references: [id])
}
```

How would you go about implementing this kind of feature?

**Comment by fnoquiq** [comment]
The self-relation documentation can be found [here](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/self-relations#many-to-many-self-relations)

I started studying prisma a week ago, but I think it would be something like this

```prisma
model Movie {
  id String @id @default(cuid())
  title String
  similar Similar[] @relation("similar")
}

model Similar {
  movieId String
  similarId String
  similar Movie @relation("similar", fields: [similarId], references: [id])

  @@id([movieId, similarId])
}
```

**Comment by ryands17** [comment, marked as answer]
@brielov 👋 

If each movie can have many recommended movies, then you would need a [1-many self relation](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/self-relations#one-to-many-self-relations) in this case.

```prisma
model Movie {
  id String @id @default(cuid())
  title String
  similarMovies  Movie[]  @relation("SimilarMovies")
  movie   Movie?   @relation("SimilarMovies", fields: [movieId], references: [id])
  movieId String?
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
